use-case-guides/workday-request-time-off/README.md

```diff
@@ -209,149 +186,346 @@ curl --location 'https://{{domain}}.workday.com/ccx/api/absenceManagement/v1/{{i
+    Returns:
+        JSON response which validates the API is working.
+    """
+    content = {"text": "Hello World"}
+    headers = {"Content-Type": "application/json"}
+    return JSONResponse(content=content, headers=headers)
+
 ```
 
+## Step 4: Build in Creator Studio
+
+We will be building this plugin as a Path along with an iPaaS to chain APIs 1, 2, and 3 together in order to submit a time off request for a user.
+
+### Plugin Architecture Diagram
+
+![Untitled](./Plugin%20Template%20Request%20Time-Off%20in%20Workday%20081c4d522bf64bbead3697288dd46047/Untitled.png)
+
+**Triggers**
+
+1. Natural language input to copilot requesting to take time off
+
+**Natural Language Slots**
+
+1. Start Date: when time off starts
+2. End Date *(optional)*: when time off ends. By default end date is the same as the start date.
+3. Comments *(optional)*: Any relevant context / justification for taking time off
+
+**Actions**
+
+1. Lookup time off type details: Executes the `/api/wql/v1/{{instance}}/data` API to query the requesting employee's Workday ID, eligible time off plans, and relevant time off types through their work email address.
+2. Request time off: Executes the `/api/absenceManagement/v1/{{instance}}/workers/{{worker_id}}/requestTimeOff` API to request time off
+
+**Caveats**
+
+1. If an employee selects a time off plan without adequante time balance, you will get an error message for attempting to request time off from the REST API. You can personalize the error message as part of exception handling logic inside your iPaaS.
+2. An employee could specify a time off plan that they are not eligible for
+1. Current Date: used in the "as of date" in the request PTO action. Could be static.
+2. Email Address: we assume that a user is only requesting time off for themselves, and use the `{{user.email_addr}}` value from user roster profile of the employee calling this plugin. This would need to be changed if an employee is requesting time off on behalf of another employee.
+
+### Build Plugin in Creator Studio
+1. Create a Path in Creator Studio. You can copy the plugin details below
+    * Plugin Name: `🏝️ Request Time Off in Workday` 
+    * Plugin Description: `Allow employees to submit time off requests in Workday.`
+    ![Name and Description](./Plugin%20Template%20Request%20Time-Off%20in%20Workday%20081c4d522bf64bbead3697288dd46047/crest-1.png)
+2. The first question should ask which time off plan the employee wants to use for their time off request.
+    * You can use **API 2** above to discover the eligible time off plans for any given employee group.
+    ![Select Time Of](./Plugin%20Template%20Request%20Time-Off%20in%20Workday%20081c4d522bf64bbead3697288dd46047/crest-2.png)
+3. Ask what the starting date for your employee's time off request.
+    * This will be configured as a Date Slot inside Creator Studio
+    ![Select Date](./Plugin%20Template%20Request%20Time-Off%20in%20Workday%20081c4d522bf64bbead3697288dd46047/crest-3.png)
```

Comment:
Let's use additional new line characters or something so it doesn't render so weird
![CleanShot 2024-07-28 at 09 33 11@2x](https://github.com/user-attachments/assets/1fee8d24-1f8d-4755-b2ef-2fa76538d269)